### PR TITLE
Set Elastic licence type for APM server Beats update job

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

Adds a new licence override temporarily when testing if Beats can be updated in APM Server.

## Why is it important?

We get the following error in the Beats update job:

```
Failed to detect licences: failed to detect licence type of /var/lib/jenkins/workspace/Beats_apm-beats-update_master/src/github.com/elastic/beats-local from /var/lib/jenkins/workspace/Beats_apm-beats-update_master/src/github.com/elastic/beats-local/LICENSE.txt:
failed to detect licence type of /var/lib/jenkins/workspace/Beats_apm-beats-update_master/src/github.com/elastic/beats-local/LICENSE.txt
```

### CODE REVIEW

**Comment by kvch** [comment]
Check APM Server Beats Update job passes: https://beats-ci.elastic.co/job/Beats/job/apm-beats-update/job/PR-24122/5/

**Review by jsoriano — APPROVED** [review]
Thanks!

File: .ci/apm-beats-update.groovy [hunk 1]
```diff
@@ -127,6 +127,7 @@ def beatsUpdate() {
         git config --global --add remote.origin.fetch "+refs/pull/*/head:refs/remotes/origin/pr/*"
 
         go mod edit -replace github.com/elastic/beats/v7=\${GOPATH}/src/github.com/elastic/beats-local
+        echo '{"name": "${GOPATH}/src/github.com/elastic/beats-local", "licenceType": "Elastic"}' >> \${GOPATH}/src/github.com/elastic/beats-local/dev-tools/notice/overrides.json
```
Comment (urso) [inline .ci/apm-beats-update.groovy:130]: playing devils advocat: We have 2 licenses. The OSS apm-server should not be build with Beats Elastic License :)

Comment (kvch) [reply .ci/apm-beats-update.groovy:130]: This is not part of the build process, and has zero impact on packaging. This is just a workaround for a job that checks if we added anything to Beats which might possibly break something in APM. Also, it is somewhat a shame that the job has been failing for more than a month and we haven't noticed it. :(

We can change it to Apache, but the text in our LICENCE.txt is not Apache. It states that it has two licences; Elastic and Apache. To me, putting Elastic in this workaround felt more truthful. I can change it regardless, but it seems unnecessary.

**Pull request merged**